sdk/core/System.ClientModel/src/Message/PipelineResponseHeaders.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Runtime.CompilerServices;
+
+namespace System.ClientModel.Primitives;
+
+internal class PipelineResponseHeaders : PipelineMessageHeaders
+{
+    private readonly HttpResponseMessage _httpResponse;
+    private readonly HttpContent _httpResponseContent;
+
+    public PipelineResponseHeaders(HttpResponseMessage response, HttpContent responseContent)
+    {
+        _httpResponse = response;
+        _httpResponseContent = responseContent;
+    }
+
+    public override void Add(string name, string value)
+        => throw new NotSupportedException();
```

Comment:
so many not supported. I am not a fan. Maybe we should not inherit from the base, or simply add a concept or read-only and r/w headers.

Reply:
That's fair.  In Azure.Core we have separate types for RequestHeaders and ResponseHeaders.  I was trying to minimize the number of redundant types in ClientModel by merging these, but if you think it's preferable to separate them back out, that's entirely doable.  In Azure.Core, having them separate also means we can add static properties for HTTP headers that are common on either request or response (but not both), which is also nice.  Let me know if you'd like to move to that.

Reply:
yeah, I think two types would be best

Reply:
I'll file a P0 issue to follow up and we'll plan to do the split before the first v1.1 beta ships.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/41230